pkg/router/listen_path_parameter_name_extractor.go

```diff
@@ -0,0 +1,30 @@
+package router
+
+import "regexp"
+
+const (
+	parameterMatchRule = `\{([^/}]+)\}`
+)
+
+// ListenPathParameterNameExtractor is responsible for extracting parameters name from the listen path
+type ListenPathParameterNameExtractor struct {
+	reg *regexp.Regexp
+}
+
+// NewListenPathParameterNameExtractor creates a new instance ListenPathParameterNameExtractor
+func NewListenPathParameterNameExtractor() *ListenPathParameterNameExtractor {
+	return &ListenPathParameterNameExtractor{regexp.MustCompile(parameterMatchRule)}
+}
+
+// Extract takes the usable part of the listen path an extracts parameter names
+func (l *ListenPathParameterNameExtractor) Extract(listenPath string) []string {
+	submatches := l.reg.FindAllStringSubmatch(listenPath, -1)
```

Comment:
What do you think about looking of `{` first and return an empty array if it's not there? Do you think it could speed up things?

Reply:
Performing on 10000 urls the difference is of 12ms on the total execution time, not sure it's worth adding a bit of complexity there, the gain seems pretty small

~64ms vs ~76ms